### PR TITLE
dsg 2351: form config behavioral subject + misc fixes

### PR DESCRIPTION
**Pull request by derekdiep42**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [x] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [x] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

The main change in this is that since we have form config creation and having form as a selection options in the same page, we should make the form configurations service store form configs as a behavioral subject to allow for dynamic updating (as observable on the most recent form configs list). 

Implements:
- Logic for form configs behavioral subject in the service, allowing the forms list to be dynamically accessed in the form config list AND the dropdown for forms int he form selection rules section
- bug fix for select component validation matches
- adding text input component valueChanges output emitter (needed for the form selection rules context)
- api route for tasks
- - - bug fix for sending empty filters in the schemas/transaction calls
- workflow task model and mocks 

## Screenshot of the affected changes

Form configs working dynamically because of observable:

https://github.com/Nuvalence/dsgov-web/assets/56942479/52233be3-2dd7-4251-abc0-d12e11ef4285

bug fix for MatchOptions not triggering right:
(before)

https://github.com/Nuvalence/dsgov-web/assets/56942479/c1858f42-3681-4b50-a08e-ce773c9c8bcc

(after)

https://github.com/Nuvalence/dsgov-web/assets/56942479/ed0a51a1-ff0b-468d-a247-5df601ca1428

empty filters fix:

Screenshot 2023-11-20 at 10 05 23 AM

Screenshot 2023-11-20 at 10 04 56 AM

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by jgnuvalence — APPROVED** [review]
LGTM

File: libs/shared/ui/nuverial/src/lib/components/text-input/text-input.component.ts [hunk 1]
```diff
@@ -177,6 +182,7 @@ export class NuverialTextInputComponent extends FormInputBaseDirective implement
 
   public ngOnInit() {
     this.formControl = this._modelFormControl();
+    this.formControl.valueChanges.pipe(tap(text => this.valueChange.emit(text))).subscribe();
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/text-input/text-input.component.ts:185]: Lets not add this until we need it

File: libs/agency/feature/admin/src/lib/components/form-configurations/form-configurations.component.ts [hunk 2]
```diff
@@ -75,8 +75,11 @@ export class FormConfigurationsComponent implements OnInit {
     this.lastUpdatedDisplay = '';
     this.createdByDisplay = '';
   }
+
   public ngOnInit(): void {
-    this.formConfigurationsList$ = this._workApiRoutesService.getFormConfigurations$(this.transactionDefinition.key).pipe(
+    this._formConfigurationService.getFormConfigurations$(this.transactionDefinition.key).pipe(take(1)).subscribe();
```
Comment (davidyoung77) [inline libs/agency/feature/admin/src/lib/components/form-configurations/form-configurations.component.ts:80]: Can we use async pipe to subscribe

**Review by davidyoung77 — COMMENTED** [review]
Thanks I have a few comments

**Review by davidyoung77 — APPROVED** [review]

**Pull request merged**